### DISCUSSION TITLE
[BP#68] Libérer de la mémoire les variables qui ne sont plus nécessaires

### DISCUSSION THREAD

**Discussion by devcrafting** [cnumr/best-practices#80] [Bonnes pratiques] [answered]
Cette BP me semble faire partie des micro-optimisations. J'ai un doute sur PHP, mais il me semble que la RAM est nettoyée à chaque fin de requête, voir même à chaque sortie de bloc (comme font les garbages collector comme en JDK ou CLR).
Cela me semble également apporter une complexification du code nuisible à la lisibilité/maintenabilité.

**Comment by rlemaire** [comment]
Oui c'est sans doute spécifique à certains langages ou technologies. Et ça me parait dans ce cas être assez basique.

**Comment by didierfred** [comment]
Oui la plupart du temps les langages intégrent un garbage collector . 
Si il faut faire qqchose , c'est travailler sur les fuites mémoires , par exemple en js : https://auth0.com/blog/four-types-of-leaks-in-your-javascript-code-and-how-to-get-rid-of-them/

Mais c'est plutôt dans le boulot de base du dev et pas forcement lié à l'eco-conception ?

**Comment by ACTLEM** [comment]
Note Point Equipe Projet: Prendre de la hauteur sur cette bonne pratique, modifier la bonne pratique pour énoncer la pertinence de la mise en cache des données (évincer les données dont on n'a plus besoin et qui sont mises en cache).

**Comment by ACTLEM** [comment, marked as answer]
Note équipe projet: A supprimer, sera remplacé par https://github.com/cnumr/best-practices/issues/176 (voir discussion https://github.com/cnumr/best-practices/discussions/47)

Issue créée: https://github.com/cnumr/best-practices/issues/177
